src/signed/tests.rs

```diff
@@ -117,28 +137,68 @@ impl MultiKeychain for TestMultiKeychain {
 }
 
 #[test]
-fn test_multisignatures() {
-    let mut msg = Indexed::new(
-        TestMessage {
-            msg: "Hello".as_bytes().to_vec(),
-        },
-        0.into(),
+fn test_valid_partial_multisignature() {
+    let index: NodeIndex = 0.into();
+    let node_count: NodeCount = 1.into();
+    let keychain = TestMultiKeychain { node_count, index };
+
+    let msg = indexed_test_message(index.0);
+    let partial = PartiallyMultisigned::sign(msg, &keychain);
+    let unchecked_msg = partial.unchecked;
+
+    assert!(
+        unchecked_msg.check_partial(&keychain).is_ok(),
+        "Partially signed message should be valid"
     );
+}
+
+#[test]
+fn test_invalid_partial_multisignature() {
+    let index: NodeIndex = 0.into();
+    let node_count: NodeCount = 1.into();
+    let keychain = TestMultiKeychain { node_count, index };
+
+    let msg = indexed_test_message(index.0);
+    let partial = PartiallyMultisigned::sign(msg, &keychain);
+    let mut unchecked_msg = partial.unchecked;
+    unchecked_msg.signature.msg = "Bye".as_bytes().to_vec();
+
+    assert!(
+        unchecked_msg.check_partial(&keychain).is_err(),
+        "Wrong signature can't pass partial check",
+    );
+}
+
+#[test]
+fn test_incomplete_multisignature() {
+    let msg = indexed_test_message(0);
+    let index: NodeIndex = 0.into();
+    let node_count: NodeCount = 2.into();
+    let keychain = TestMultiKeychain { node_count, index };
+
+    let partial = PartiallyMultisigned::sign(msg, &keychain);
+    assert!(
+        partial._try_into_complete(&keychain).is_err(),
+        "5 signatures should form a complete signature"
```

Comment:
Probably a message "1 signature does not form a complete signature" is more descriptive?

Reply:
Definitely.